cmd/soroban-cli/src/commands/network/start.rs

```diff
@@ -0,0 +1,151 @@
+use std::process::{Command, Stdio};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {}
+
+const FROM_PORT: i32 = 8000;
+const TO_PORT: i32 = 8000;
+const CONTAINER_NAME: &str = "stellar";
+const DOCKER_IMAGE: &str = "stellar/quickstart";
+
+/// This command allows for starting a stellar quickstart container. To run it, you can use the following command:
+/// `soroban network start <NETWORK> [OPTIONS] -- [DOCKER_RUN_ARGS]`
+///
+/// OPTIONS: refer to the options that are available to the quickstart image:
+/// --enable-soroban-rpc - is enabled by default
+/// --protocol-version (only for local network)
+/// --limits (only for local network)
+
+/// `DOCKER_RUN_ARGS`: These are arguments to be passed to the `docker run` command itself, and should be passed in after the slop `--`. Some common options are:
+/// -p <`FROM_PORT`>:<`TO_PORT`> - this maps the port from the container to the host machine. By default, the port is 8000.
+/// -d - this runs the container in detached mode, so that it runs in the background
+
+// By default, without any optional arguments, the following docker command will run:
+// docker run --rm -p 8000:8000 --name stellar stellar/quickstart:testing --testnet --enable-soroban-rpc
+
+#[derive(Debug, clap::Parser, Clone)]
+pub struct Cmd {
+    /// Network to start, e.g. local, testnet, futurenet, pubnet
+    pub network: String,
```

Comment:
As written, this allows passing any string for the `network` argument. I wonder if we should make this an `enum` and hard-code an allowed list. Not sure what I would expect it to do, if I wrote something like `soroban network start lolfactory`. 

I also kinda like the idea of using this as the `--name` argument to `docker run`, so that the commands become parallel. `soroban network start local` / `soroban network stop local`. 

Not sure where to say this next bit, but it's sorta related to this, so I'll put it here for now: I also think we should start in detached mode by default. BUT, with the modification that we only "release" back to the prompt after the container is _healthy_ — once RPC and Friendbot are up and running. 

Right now, when I start a network in detached mode, I see this:

<img width="1010" alt="image" src="https://github.com/stellar/soroban-tools/assets/221614/180e0efc-850d-4231-99d5-10b099a2c55c">

Not sure what that `0f50...` output is.

In my ideal version, we'd show a [loading animation](https://lib.rs/crates/spinners) and a message like "waiting for quickstart to fully boot...". It might be worth copying colima's approach here, where you can see the full log of what's happening scroll by in a limited ~7 line space, but greyed-out . Once RPC and Friendbot are ready, then we output a message like:

```
Stellar local quickstart container is up and running!
  Soroban RPC: http://localhost:8000/soroban/rpc
  Friendbot: http://localhost:8000/friendbot
  Horizon: ...
  etc: ...
  etc: ... [do we need more than RPC and Friendbot??]
To stop this container: `soroban network stop local`
```

And drop them back to their prompt.

Reply:
* > As written, this allows passing any string for the network argument. I wonder if we should make this an enum and hard-code an allowed list. Not sure what I would expect it to do, if I wrote something like soroban network start olfactory.

    Good call - this should be an enum! Updated in 8457ecbd1000098a71673816b931f604e2754113

* > I also kinda like the idea of using this as the --name argument to docker run, so that the commands become parallel. soroban network start local / soroban network stop local.

    Ah, interesting idea! I like `soroban network start local` and `soroban network stop local` being the same! 
  
    If we are prioritizing the commands being the same, I'll remove the ability for a user to pass in their own container name for now (`--container-name`), since it complicates having `soroban network start local` and `soroban network stop local` be in sync if the user is making up their own docker container name.

Update in [6687e37](https://github.com/stellar/soroban-tools/pull/1107/commits/6687e378570c9005465ab6b1b1d4192131606280)

Reply:
I didn't get a chance to respond to the other part your comment last week, but here it is!
> I also think we should start in detached mode by default. BUT, with the modification that we only "release" back to the prompt after the container is healthy — once RPC and Friendbot are up and running....

This is a great call! Once switching to using bollard to handle the docker interaction, it seems to default to run in detached mode. So I am thinking that I'll keep with that, and then in a future PR we can add the ability to have it be non-detached mode, and stream the output of the container.. if we decide that's important.

I love the idea of running a healthcheck, and having some more helpful messaging, instead of just spitting out the container id, which it was it used to do. I've made a small improvement, but I’m leaning toward pushing the health check piece to a follow-up PR so we can get this one in.

It now looks like this when the container is started successfully:
```
ℹ️  Starting testnet network
✅ Container started: stellar-testnet
ℹ️  To stop this container run: soroban network stop testnet --docker-socket-path /Users/elizabethengelman/.docker/run/docker.sock
```

Reply:
Marking this comment thread as resolved, and have added some follow up tasks in https://github.com/orgs/stellar/projects/50?pane=issue&itemId=54158722